core/consensus/utils/instance_io.go

```diff
@@ -17,23 +18,20 @@ const (
 // NewInstanceIO returns a new instanceIO.
 func NewInstanceIO[T any]() *InstanceIO[T] {
 	return &InstanceIO[T]{
-		Participated: make(chan struct{}),
-		Proposed:     make(chan struct{}),
-		Running:      make(chan struct{}),
-		RecvBuffer:   make(chan T, RecvBufferSize),
-		HashCh:       make(chan [32]byte, 1),
-		ValueCh:      make(chan proto.Message, 1),
-		ErrCh:        make(chan error, 1),
-		DecidedAtCh:  make(chan time.Time, 1),
+		RecvBuffer:  make(chan T, RecvBufferSize),
+		HashCh:      make(chan [32]byte, 1),
+		ValueCh:     make(chan proto.Message, 1),
+		ErrCh:       make(chan error, 1),
+		DecidedAtCh: make(chan time.Time, 1),
 	}
 }
 
 // InstanceIO defines the async input and output channels of a
 // single consensus instance in the Component.
 type InstanceIO[T any] struct {
-	Participated chan struct{}      // Closed when Participate was called for this instance.
-	Proposed     chan struct{}      // Closed when Propose was called for this instance.
-	Running      chan struct{}      // Closed when runInstance was already called.
+	Participated int32              // Closed when Participate was called for this instance.
```

Comment:
We should change those comments, there are no channels involved in this anymore.

Reply:
Why not an `atomic.Bool`?

Reply:
Indeed, changed as suggested.